### PR TITLE
Draft: improve some extended indexes documentation

### LINKED ISSUE #3332: Post-release improvements to extended indexes docs
A couple of good suggestions on [the extended indexes PR](https://github.com/prisma/docs/pull/3235) that I'm not going to get to before the release:

- https://www.prisma.io/docs/concepts/components/prisma-schema/indexes#index-configuration : find a nicer way to organise information - better ordering of database, version and argument information
- https://www.prisma.io/docs/reference/database-reference/database-features#indexes Include Prisma syntax somehow (e.g. `type` argument)

### PR DESCRIPTION
**Pull request by keerlu**
## What issue does this fix?

Fixes #3332

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx [hunk 1]
```diff
@@ -24,13 +24,22 @@ If you are upgrading from a version earlier than 4.0.0, these changes to index c
 
 You can configure indexes, unique constraints, and primary key constraints with the following attribute arguments:
 
-- The `length` argument is available **in MySQL only** on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` attributes in version `3.5.0` and later. It allows Prisma to support indexes and constraints on `String` and `Bytes` types.
+- The `length` argument allows you to specify a maximum length for the subpart of the value to be indexed on `String` and `Bytes` types
 
-- The `sort` argument is available for all databases on the `@unique`, `@@unique` and `@@index` attributes in version `3.5.0` and later. Additionally, SQL Server also allows it on `@id` and `@@id`.
+  - Available on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` attributes **in MySQL only**, in versions 3.5.0 and later
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx]: Have you tried splitting this bullet into 3? Will cost vetrical space, but sperate the different information bits:
```suggestion
  - Available on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` attributes 
  - **MySQL only**
  - Versions 3.5.0 and later
```
or similar

File: content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx [hunk 2]
```diff
@@ -24,13 +24,22 @@ If you are upgrading from a version earlier than 4.0.0, these changes to index c
...
+- The `sort` argument allows you to specify the order that the entries of the constraint or index are stored in the database
 
-- The `clustered` argument is available **in SQL Server only** on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` attribute in version `3.13.0` and later. It allows Prisma to configure (non)clustered indexes.
+  - Available on the `@unique`, `@@unique` and `@@index` attributes in all databases and on the `@id` and `@@id` attributes in SQL Server, in versions 3.5.0 and later
+
+- The `type` argument allows you to support index access methods other than PostgreSQL's default `BTree` access method
+
+  - Available on the `@@index` attribute **in PostgreSQL only**
+  - The `Hash` index access method is available in versions 3.6.0 and later
+  - Additionally the `Gist`, `Gin`, `SpGist` and `Brin` access methods are available in versions 3.14.0 and later
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx]: Similar to something I commented in Slack:
I think the version information really does not need to be present here, and can be part of the detailled section below.

File: content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx [hunk 3]
```diff
@@ -24,13 +24,22 @@ If you are upgrading from a version earlier than 4.0.0, these changes to index c
...
+
+- The `clustered` argument allows you to configure whether a constraint or index is clustered or non-clustered
+  - Available on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` attributes **in SQL Server only**, in version 3.13.0 and later. It allows Prisma to configure (non)clustered indexes.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx]: This information would not be duplicated, so removed it:
```suggestion
  - Available on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` attributes **in SQL Server only**, in version 3.13.0 and later.
```

File: content/400-reference/300-database-reference/01-database-features.mdx [hunk 4]
```diff
@@ -56,11 +56,11 @@ This section describes which database features exist on the relational databases
...
+| -------- | ---------- | -------------------- | ----- | ------ | ----------- | ---------------------------------------------------------------------------------------------------------------------------- | ------------- | -------------- |
+| `UNIQUE` | ✔️         | ✔️                   | ✔️    | ✔️     | ✔️          | [`@unique` and `@@unique`](/concepts/components/prisma-schema/data-model#defining-a-unique-field)                            | ✔️            | ✔️             |
+| `WHERE`  | ✔️         | ✔️                   | ✔️    | ✔️     | ✔️          | Not yet                                                                                                                      | ✔️            | Not yet        |
+| `USING`  | ✔️         | ✔️                   | ✔️    | ✔️     | ✔️          | [`type`](/concepts/components/prisma-schema/indexes#configuring-the-access-type-of-indexes-with-type-postgresql)<sup>†</sup> | ✔️            | Not yet        |
 
 - † Available with the PostgreSQL connector only in Prisma versions `4.0.0` and later.
```
Comment (janpio) [inline content/400-reference/300-database-reference/01-database-features.mdx]: Semi (un)related: Is this version correct?

**Review by janpio — APPROVED** [review]
Looks much better already.

Left some small fix and suggestions as comment.

File: content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx (hunk 3 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx]: Thanks, got missed when I was copypasting the pieces around.

File: content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx (hunk 1 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx]: Thanks, have tried this (see latest commit) and I think I prefer it

File: content/400-reference/300-database-reference/01-database-features.mdx (hunk 4 above)
Comment (keerlu) [reply content/400-reference/300-database-reference/01-database-features.mdx]: It's correct if we're talking about GA, but not preview. So potentially better to have:

```suggestion
- † Available in preview in 3.6.0 and later and in general availability in 4.0.0 and later, with the PostgreSQL connector only.
```

**Review by janpio — APPROVED** [review]

**Pull request merged**